### PR TITLE
geopandas 0.12.2

### PR DESCRIPTION
**Pull request by skupr-anaconda**
**Jira ticket:** [PKG-1047](https://anaconda.atlassian.net/browse/PKG-1047)

**The upstream data:**

Changelog: https://github.com/geopandas/geopandas/blob/v0.12.2/CHANGELOG.md
License: https://github.com/geopandas/geopandas/blob/v0.12.2/LICENSE.txt
Requirements:
- https://github.com/geopandas/geopandas/blob/v0.12.2/pyproject.toml
- https://github.com/geopandas/geopandas/blob/v0.12.2/setup.py

**_Actions:_**

- Update **source/url**

- Skip `py<38` and `py>=310`, seems like there is no support for **python 3.10** at all

- Skip `s390x` because of missing `rtree`

- Add `build_base.bat`  # [win] 

- Fix `python` in `host` and `run`

- Add missing `wheel` and `setuptools` to `host`

- Update pinnings in `run` and add `folium` to geopandas' `run` section

- Update `test/imports`
- Update `run_test.py` for testing all available file formats in the `test_data` folder

- Fix `home` url

- Add `license_family`

- Add `description`

- Update dev and doc urls

**_Notes:_**
 * geopandas 0.12.2 assigns Crimea to Ukraine in the `naturalearth_lowres` built-in dataset

**Additional info:**

<details>

**Package's statistics**

<details>

 * Priority C | effort: easy | Category: anaconda_affiliated_deps | subcategory:  | pkg_type: python
 * Outdated platfroms: 'linux-64', 'linux-aarch64', 'linux-ppc64le', 'osx-64', 'osx-arm64', 'win-64'
 * Latest version: 0.12.2
 * Release on PyPi:
    * version: 0.12.2
    * date: 2022-12-10T18:39:09
* [PyPi history](https://pypi.org/project/geopandas/#history)
 * Popularity: 
    * 3 months downloads: 152030.0
    * All time:  2286218 downloads -  geopandas  0.12.2  Geographic pandas extensions  

</details>

**Other checks:**

<details>
        
8. - [x] Verify that the `build_number` is correct
9. - [x] Verify if the package needs `setuptools`
    
10. - [x] Verify if the package needs `wheel`
    
11. - [x] `pip` in test
    
12. - [x] Verify the test section
13. - [x] Verify if the package is `architecture specific`
14. - [x] Verify that private modules are not mentioned in the recipe For example: (_private_module)
15.  - [x] license_file: LICENSE.txt is present

16. - [x] License family is present
    
17. - [x] License: BSD-3-Clause
    - [x] License is `spdx` compliant
 * Check if the license identifier has correct name from the SPDX License List

| Identifier                           |
|:-------------------------------------|
| BSD-3-Clause                         |
| BSD-3-Clause-Attribution             |
| BSD-3-Clause-Clear                   |
| BSD-3-Clause-LBNL                    |
| BSD-3-Clause-Modification            |
| BSD-3-Clause-No-Military-License     |
| BSD-3-Clause-No-Nuclear-License      |
| BSD-3-Clause-No-Nuclear-License-2014 |
| BSD-3-Clause-No-Nuclear-Warranty     |
| BSD-3-Clause-Open-MPI                |
</details>

**Check dependency issues:**

<details>
../aggregate/geopandas-feedstock/recipe/meta.yaml
Dependencies: ['mapclassify >=2.2.0', 'python >=3.6>=3.6', 'shapely', 'fiona', 'rtree', 'matplotlib-base', 'pandas >=0.24', 'pyproj >=2.2.0', 'geopandas-base', 'pip']

</details>

**Links:**
* [Anaconda Recipes feedstock](https://github.com/AnacondaRecipes/geopandas-feedstock)
* [Update branch](https://github.com/AnacondaRecipes/geopandas-feedstock/tree/0.12.2)
* [conda-forge recipe](https://github.com/{upstream}/{feedstock_name}-feedstock)
* [PyPI](https://pypi.org/project/geopandas)
* [Diff between upstream and feature branch](https://github.com/conda-forge/geopandas-feedstock/compare/main...AnacondaRecipes:0.12.2)
* [Diff between origin and feature branch](https://github.com/AnacondaRecipes/geopandas-feedstock/compare/master...AnacondaRecipes:0.12.2)
* [The last merged PRs](https://github.com/AnacondaRecipes/geopandas-feedstock/pulls?q=is%3Apr+is%3Amerged+sort%3Aupdated-desc+)

</details>

**Updating the recipe:**
If the recipe needs additional modification the update branch can be modified. Note that the PR diffs are not updated with these changes.
```
git clone -b 0.12.2 git@github.com:AnacondaRecipes/geopandas-feedstock.git
```

[PKG-1047]: https://anaconda.atlassian.net/browse/PKG-1047?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by skupr-anaconda** [comment]
An error on `s390x`:
```
conda.exceptions.ResolvePackageNotFound:
  - rtree
  - fiona[version='>=1.8']
  - geopandas-base==0.12.2=py310ha847dfd_0
  - xyzservices
```
We can skip building on `s390x`

**Comment by skupr-anaconda** [comment]
`geopandas` dependencies cause errors for **python 3.10**. It seems like one of them is missing artifacts

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,63 +1,92 @@
...
+        - setuptools
+        - wheel
       run:
-        - python >=3.6
+        - python
         - {{ pin_subpackage('geopandas-base', exact=True) }}
-        - fiona
-        - rtree
+        - fiona >=1.8
+        # fiona package isn't available yet on the defaults channel.
```
Comment (ViridianMelody) [inline recipe/meta.yaml]: Fiona is now on defaults (now including Python 3.10) for all but s390x, so you should be able to build this now.

Comment (skupr-anaconda) [reply recipe/meta.yaml]: Thanks Melody!

**Comment by skupr-anaconda** [comment]
geopandas has unsatisfiable dependencies for python 3.10 (Seems like the incompatible dependencies are
- libgdal-3.0.2-h0f233a1_0 requires proj >=6.2.1,<6.2.2.0a0, but we have pyproj with a dependent proj >=8.2.1). 

Generally, it's not clear if geopandas 0.12.2 has support for python 3.10 on all platforms, see https://github.com/geopandas/geopandas/blob/v0.12.2/.github/workflows/tests.yaml

Skip building for python 3.10

**Review by ELundby45 — APPROVED** [review]
LGTM!

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,63 +1,96 @@
...
+        - setuptools
+        - wheel
       run:
-        - python >=3.6
-        - pandas >=0.24
-        - shapely
-        - pyproj >=2.2.0
+        - python
+        - packaging
+        - pandas >=1.0.5
```
Comment (cbouss) [inline recipe/meta.yaml]: https://github.com/geopandas/geopandas/blob/v0.12.2/setup.py#L36
```suggestion
        - pandas >=1.0.0
```

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,63 +1,96 @@
...
+        - pyproj >=2.6.1.post1
+        - shapely >=1.7
```
Comment (cbouss) [inline recipe/meta.yaml]: I don't see fiona as optional.
https://github.com/geopandas/geopandas/blob/v0.12.2/setup.py#L38
```suggestion
        - fiona >=1.8
```

File: recipe/meta.yaml [hunk 4]
```diff
@@ -1,63 +1,96 @@
...
       run:
-        - python >=3.6
+        - python
         - {{ pin_subpackage('geopandas-base', exact=True) }}
-        - fiona
-        - rtree
+        - fiona >=1.8
+        - folium
+        - mapclassify >=2.4.0
         - matplotlib-base
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
        - matplotlib-base >=3.2.0
```
See: https://geopandas.org/en/v0.12.2/getting_started/install.html#dependencies

File: recipe/meta.yaml [hunk 5]
```diff
@@ -1,63 +1,96 @@
...
+        - pip
+        - setuptools
+        - wheel
       run:
-        - python >=3.6
+        - python
         - {{ pin_subpackage('geopandas-base', exact=True) }}
-        - fiona
-        - rtree
+        - fiona >=1.8
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
```
As it's a dependency for geopandas-base

File: recipe/meta.yaml [hunk 6]
```diff
@@ -1,63 +1,96 @@
...
+        - folium
+        - mapclassify >=2.4.0
         - matplotlib-base
-        - mapclassify >=2.2.0
+        - rtree
+        - xyzservices
```
Comment (cbouss) [inline recipe/meta.yaml:62]: xyzservices and folium are not mentioned in geopandas, but are ok to add given it was suggested in the CF PR: https://github.com/conda-forge/geopandas-feedstock/pull/93

**Review by ViridianMelody — APPROVED** [review]
I looked into the Python 3.10 dependency issues a bit, and to rebuild `gdal` with `proj 8.2.1` we'd need to update `libspatialite` and `geotiff` at the least. With `geopandas` not definitively supporting 3.10 yet, I agree it's probably best we wait for a future release.

File: recipe/meta.yaml (hunk 5 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml]: Thanks. I moved it to geopandas-base

File: recipe/meta.yaml [hunk 7]
```diff
@@ -1,63 +1,95 @@
...
 build:
-  number: 1
-  noarch: python
+  number: 0
+  # rtree is missing on s390x.
+  # geopandas has unsatisfable dependencies for python 3.10 (Seems like the incompatible dependencies are
+  # libgdal-3.0.2-h0f233a1_0 requires proj >=6.2.1,<6.2.2.0a0, but we have pyproj with a dependent proj >=8.2.1). 
+  # Generaly, it's not clear if geopandas 0.12.2 has support for python 3.10 on all platfroms,
+  # see https://github.com/geopandas/geopandas/blob/v0.12.2/.github/workflows/tests.yaml
+  skip: True  # [py<38 or s390x or py>=310]
```
Comment (jorisvandenbossche) [inline recipe/meta.yaml:19]: Of course we support Python 3.10 (geopandas is a pure python package, so this is probably an issue with the dependencies on the defaults channel). Could this skip be removed?

Comment (skupr-anaconda) [reply recipe/meta.yaml:19]: Yeah, there was an incompatibility of `libgdal-3.0.2-h0f233a1_0` that requires `proj >=6.2.1,<6.2.2.0a0`, but we have `pyproj` with a dependent `proj >=8.2.1`. We should first rebuild/update `libgdal` to make it compatible with `pyproj`/`proj`

**Pull request merged**